ConfidenceDemoApp/src/main/java/com/example/confidencedemoapp/MainVm.kt

```diff
@@ -30,51 +23,39 @@ class MainVm(app: Application) : AndroidViewModel(app) {
         private val TAG = MainVm::class.java.simpleName
     }
 
-    private var client: Client
-    private var ctx: EvaluationContext = ImmutableContext(targetingKey = "a98a4291-53b0-49d9-bae8-73d3f5da2070")
     private val _message: MutableLiveData<String> = MutableLiveData("initial")
     private val _color: MutableLiveData<Color> = MutableLiveData(Color.Gray)
     val message: LiveData<String> = _message
     val color: LiveData<Color> = _color
     private var eventSender: EventSender
+    private var confidence: Confidence
 
     init {
         val start = System.currentTimeMillis()
         val clientSecret = ClientSecretProvider.clientSecret()
 
-        val strategy = if (ConfidenceFeatureProvider.isStorageEmpty(app.applicationContext)) {
-            InitialisationStrategy.FetchAndActivate
-        } else {
-            InitialisationStrategy.ActivateAndFetchAsync
-        }
-
-        client = OpenFeatureAPI.getClient()
-
-        val values = Value.List(listOf(Value.String(""), Value.Double(2.0)))
-
-
         val mutableMap = mutableMapOf<String, ConfidenceValue>()
         mutableMap["screen"] = ConfidenceValue.String("value")
         mutableMap["hello"] = ConfidenceValue.Boolean(false)
         mutableMap["NN"] = ConfidenceValue.Double(20.0)
         mutableMap["list"] = ConfidenceValue.stringList(listOf(""))
         mutableMap["my_struct"] = ConfidenceValue.Struct(mapOf("x" to ConfidenceValue.Double(2.0)))
 
-        val confidence = ConfidenceFactory.create(
+        confidence = ConfidenceFactory.create(
             app.applicationContext,
             clientSecret,
+            initialContext = mapOf("targeting_key" to ConfidenceValue.String("a98a4291-53b0-49d9-bae8-73d3f5da2070")),
```

Comment:
Would love for us to move away from `targeting_key`. Do we keep it for the flag rules in the backend expect it?

Reply:
I'm not sure. i think backend works without targeting key.